### ISSUE TITLE
Cant import ray after installing latest wheel (MacOS)

### ISSUE THREAD

**Issue by MoggeM** [ray-project/ray#2763]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS 10.12.6
- **Ray installed from (source or binary)**: Pip wheel
- **Ray version**: 0.5.0 (Wheel)
- **Python version**: 3.6
- **Exact command to reproduce**:

### Describe the problem
Yesterday I installed the latest wheel from pip and now im getting an error importing ray. I had no problems with the latest wheel 1-2 weeks ago on the same platform

### Source code / logs
import ray

----> 1 import ray

~/anaconda3/lib/python3.6/site-packages/ray/__init__.py in <module>()
     47     raise
     48 
---> 49 from ray.local_scheduler import ObjectID, _config  # noqa: E402
     50 from ray.profiling import profile  # noqa: E402
     51 from ray.worker import (error_info, init, connect, disconnect, get, put, wait,

~/anaconda3/lib/python3.6/site-packages/ray/local_scheduler/__init__.py in <module>()
      3 from __future__ import print_function
      4 
----> 5 from ray.core.src.local_scheduler.liblocal_scheduler_library_python import (
      6     Task, LocalSchedulerClient, ObjectID, check_simple_value, compute_task_id,
      7     task_from_string, task_to_string, _config, common_error)

ImportError: dlopen(/Users/morelf/anaconda3/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so, 2): Symbol not found: _syslog$DARWIN_EXTSN
  Referenced from: /Users/morelf/anaconda3/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so (which was built for Mac OS X 10.13)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/morelf/anaconda3/lib/python3.6/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so

**Comment by robertnishihara** [comment]
Does the issue occurs in the following settings?
1. When you install `pip install -U ray==0.5.0`
2. When you install `pip install -U ray==0.5.1`
3. When you install `pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.5.1-cp36-cp36m-macosx_10_6_intel.whl`

**Comment by MoggeM** [comment]
pip install -U ray==0.5.0 works
pip install -U ray==0.5.1 does not work, throws the same error
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.5.1-cp36-cp36m-macosx_10_6_intel.whl does not work, throws the same error

**Comment by robertnishihara** [comment]
@pcmoritz any idea about this? I built the wheels for `0.5.1` on my laptop (macOS 10.13). The wheels in the S3 bucket were built in Travis (currently also 10.13).

I can't remember where the wheels for `0.5.0` were built.

**Comment by old-bear** [comment]
Updated to the latest version and got the same error on my MacOS 10.11:
```
ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so, 2): Symbol not found: _syslog$DARWIN_EXTSN
  Referenced from: /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ray/core/src/local_scheduler/liblocal_scheduler_library_python.so (which was built for Mac OS X 10.13)
``` 
The former build seems to support lower MacOS version. Hope we can still have this backward compatibility because it's very inconvenient to upgrade the macos version...

Thanks

**Comment by MoggeM** [comment]
@old-bear I managed to get it working by installing directly from github instead. Be sure to check https://github.com/ray-project/ray/issues/1719 if you run into any problems installing, I did and that issue solved it.

**Comment by old-bear** [comment]
@MoggeM Unfortunately, Installation from the source can't help under my situation. My project is yet another library that depends on ray. It's impossible to ask all the users of my project to install ray in this way.

**Comment by atumanov** [comment]
@old-bear @MoggeM , we've built the replacement wheels and are in the process of testing them as we speak. We should be able to push them in the matter of a couple of hours. I've just heard confirmation that the cp36 macosx wheel I built last night is working on 10.13.3 (as well as 10.11). Please let us know if you need the wheels sooner.

**Comment by robertnishihara** [comment]
@MoggeM can you see if the attached wheel works (you'll need to rename it to remove the `.zip`).

[ray-0.5.1-cp36-cp36m-macosx_10_6_intel.whl.zip](https://github.com/ray-project/ray/files/2333165/ray-0.5.1-cp36-cp36m-macosx_10_6_intel.whl.zip)

**Comment by robertnishihara** [comment]
@MoggeM Can you try `pip install -U ray==0.5.2` and see if that works?

**Comment by robertnishihara** [comment]
@MoggeM @atumanov can you check the following?

Does the error happen with the following (this is the commit that introduced google logging):
```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/eec1a3eb89a40d7b2a1efd42db1d2e687e4e4a27/ray-0.5.0-cp36-cp36m-macosx_10_6_intel.whl
```

Does the error happen with the following (the commit right before google logging was introduced):
```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/4be324efc36f5f0cd66ba2f6d81bb39fef258e3e/ray-0.5.0-cp36-cp36m-macosx_10_6_intel.whl
```

**Comment by old-bear** [comment]
@robertnishihara 0.5.2 works now on MacOS 10.11
Error still happens using this (the one introduced google logging):
```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/eec1a3eb89a40d7b2a1efd42db1d2e687e4e4a27/ray-0.5.0-cp36-cp36m-macosx_10_6_intel.whl
```

And no error for that (the one before glog):
```
pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/4be324efc36f5f0cd66ba2f6d81bb39fef258e3e/ray-0.5.0-cp36-cp36m-macosx_10_6_intel.whl

```

**Comment by guoyuhong** [comment]
I've test `-mmacosx-version-min=10.12` on Python2.7. It worker.  I will submit a PR. And I will build two Python3.6 Wheel for @old-bear to verify.

**Comment by guoyuhong** [comment]
Hi @old-bear , could you please do `git clone https://github.com/guoyuhong/resources`. And verify the wheel is working?

`https://github.com/guoyuhong/resources/tree/master/dist_no_fix` should raise the same error as this issue described. 

`https://github.com/guoyuhong/resources/tree/master/dist_with_fix` should have fixed the problem.

Thanks!

**Comment by robertnishihara** [comment]
@atumanov could you try the instructions in the comment above and see if it works for you on Mac 10.11?

**Comment by robertnishihara** [comment]
@guoyuhong you built the above wheels on MacOS 10.13, right?

**Comment by guoyuhong** [comment]
@robertnishihara Yes, these two wheel files are built in MacOS 10.13. Before I submit the PR, I test the PR using @joneswong 's Mac which is 10.11, but the python version is 2.7.

**Comment by guoyuhong** [comment]
I want to make sure this solve @old-bear 's problem.

**Comment by old-bear** [comment]
@guoyuhong Thanks, the above wheels work as expected on my MacOS 10.11

**Comment by guoyuhong** [comment]
@old-bear Sorry to bother again. I have a comparison wheel file. `https://github.com/guoyuhong/resources/tree/master/dist_no_fix` should not work and `https://github.com/guoyuhong/resources/tree/master/dist_with_fix` should work, right?

**Comment by atumanov** [comment]
I am getting the same results as @old-bear on macosx10.11/py36

**Comment by guoyuhong** [comment]
Thanks, @old-bear @atumanov for the confirmation!

**Comment by robertnishihara** [comment]
Fixed by #2775.
